### PR TITLE
[skip circleci] Update release instructions

### PR DESCRIPTION
**Pull request by jneira**
* Adding prerelease sanity checklist
* Mention gitlab repo and its ci
* The page is a litle bit messed, tempted to rewrite it completely. Hope @michaelpj will have some insights to improve it

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/2384"></a>

### CODE REVIEW

**Comment by jneira** [comment]
The build check has been succesfull in my repo: https://github.com/jneira/haskell-language-server/actions/runs/1489172293
Artifacts can be chacjed here https://github.com/jneira/haskell-language-server/actions/runs/1489172293/artifacts

File: docs/contributing/releases.md [hunk 1]
```diff
@@ -16,23 +16,38 @@ and it is being used in nix environments.
...
-* [ ] check uploaded binaries (see windows note below) and the release description (usually the changelog entry) and uncheck the prerelease box
-* [ ] make public the release in the usual social channels: irc, twitter, reddit, discord, discourse, mailing lists, etc (not required but useful to spread the word :slightly_smiling_face:)
-
+- [ ] add that list to the actual [Changelog](https://github.com/haskell/haskell-language-server/blob/master/ChangeLog.md) with a description of the release.
+- [ ] bump up versions of changed packages. All are optional but [haskell-language-server itself](https://github.com/haskell/haskell-language-server/blob/master/haskell-language-server.cabal).
+- [ ] create the tag and make an initial prerelease to trigger the ci workflow (see details below)
+- [ ] contact ghcup team (#haskell-ghcup irc channel or via its [repo](https://gitlab.haskell.org/haskell/ghcup-hs/-/issues)) to try to sync our release and its inclusion in ghcup
+- [ ] check uploaded binaries (see windows note below) and the release description (usually the changelog entry) and uncheck the prerelease box
```
Comment (michaelpj) [inline docs/contributing/releases.md:32]: which uploaded binaries? where are they? I guess they were generated somewhere based on the previous workflows?

File: docs/contributing/releases.md [hunk 2]
```diff
@@ -16,23 +16,38 @@ and it is being used in nix environments.
...
+  - [ ] discord
+  - [ ] discourse
+  - [ ] reddit
 ### hackage release
 
-* [ ] bump up package versions following the [pvp specification](https://pvp.haskell.org/) if they are not already updated
-* [ ] create ${version}-hackage branch to trigger the hackage github workflow which will upload all changed packages to hackage as candidates
-* [ ] check manually candidates in hackage
-* [ ] publish them definitely
+- [ ] bump up package versions following the [pvp specification](https://pvp.haskell.org/) if they are not already updated. You could use [policeman](https://github.com/kowainik/policeman) to help with this step.
```
Comment (michaelpj) [inline docs/contributing/releases.md:42]: how is this different to the previous verison-bumping step on line 29? should there just be one version-bumping step done before any releasing?

File: docs/contributing/releases.md [hunk 3]
```diff
@@ -140,11 +155,20 @@ To manually upload a new binary we should:
...
-if the package version in the branch is different from hackage.
-If the package in the branch has the same version as the released one,
-it will check the relevant files have not changed and will throw an error
-otherwise.
+That script checks, generates the tar.gz files, unpacks and builds them in isolation
+against hackage head if the package version in the branch is different from hackage.
+If the package in the branch has the same version as the released one, it will check
+the relevant files have not changed and will throw an error otherwise.
+
+You can trigger a build only doing the above step pushing a branch named `*-check-hackage`.
```
Comment (michaelpj) [inline docs/contributing/releases.md:163]:
```suggestion
You can trigger a build which only does the above step by pushing a branch named `${version}-check-hackage`.
```

File: docs/contributing/releases.md [hunk 4]
```diff
@@ -140,11 +155,20 @@ To manually upload a new binary we should:
...
+
+The script will upload the tarballs as candidates, maintainers will have to check and publish them definitely.
+
+## haskell gitlab release pipeline
 
-The script will upload the tarballs as candidates, maintainers will have
-to check and publish them definitely.
+The project is present in the haskell gitlab server: https://gitlab.haskell.org/haskell/haskell-language-server
+The main motivation is leverage the ci infrastructure which includes architectures not included in the github ci.
```
Comment (michaelpj) [inline docs/contributing/releases.md:170]:
```suggestion
The main motivation is to leverage the ci infrastructure which includes architectures not included in the github ci.
```

**Review by michaelpj — APPROVED** [review]
I agree it's a pretty ferocious page, but it seems like there just is a lot to do! I don't have any great ideas for restructuring it.

File: docs/contributing/releases.md (hunk 1 above)
Comment (jneira) [inline docs/contributing/releases.md:32]: maybe it is a little bit clearer with

```suggestion
- [ ] in the github release edit page, check the attached binaries and the release description (usually the changelog entry) and uncheck the prerelease box
```

File: docs/contributing/releases.md (hunk 2 above)
Comment (jneira) [reply docs/contributing/releases.md:42]: The difference is in the `All are optional but haskell-language-server itself` part. For hackage bump up versions following pvp is mandatory. For the github release you only need to bump up hls itself.
At the beginning we did not have the github and hackage release in sync and we only bumped up plugin and package versions in a systematic way for the hackage one. In last releases i bumped up versions before start the release engine.

**Comment by jneira** [comment]
> I agree it's a pretty ferocious page, but it seems like there just is a lot to do! I don't have any great ideas for restructuring it.

I see some repetition between the check list and the descriptions below but maybe it is necessary. Anyways it could be rewritten in an postrior pr.

**Pull request merged**